### DISCUSSION TITLE
Datagrid PageSize auto reset when a row is clicked

### DISCUSSION THREAD

**Discussion by salucien** [Megabit/Blazorise#2442] [Q&A] [answered]
Hello !
I have a datagrid with RowDoublecliked event and ShowPageSize on true. 
```
<DataGrid TItem="Commandetest"
                      Data="@commandes"
                      RowStyling="OnRowStyling"
                      PageSize="5"
                      @bind-SelectedRow="@selectedCommand"
                      RowDoubleClicked="(()=>OnSelectedRowChanged(selectedCommand))"
                      ShowPager="true"
                      PagerPosition="DataGridPagerPosition.TopAndBottom"
                      Narrow="true"
                      ShowPageSizes="true"
                      Resizable="true"
                      Responsive="true">
```
When I try to double click on a row to trigger the event, the page size changes to 5 (the default value) on the first click if a another page size has been selected.
What is wrong with my datagrid declaration ?

**Comment by stsrki** [comment, marked as answer]
It's because you have `PageSize="5"`. Blazor works like that. It will reset it every time you refresh something on a page. Either remove that line or make it like `@bind-Page="@yourPageNumberVariable"`

**Reply by salucien** [reply]
Oh ! 
It works now !
Thank you 😄
